src/Tools/CoreFx.Tools/GenerateConfigurationProps.cs

```diff
@@ -111,6 +119,32 @@ private void ParseConfigurationString(string propertyName, ProjectRootElement pr
             }
         }
 
+        private void CreateRuntimeIdentifier(ProjectRootElement project)
+        {
+            var rid = Microsoft.DotNet.PlatformAbstractions.RuntimeEnvironment.GetRuntimeIdentifier();
+
+            string[] ridParts = rid.Split('-');
+
+            if (ridParts.Length < 1)
+                throw new System.InvalidOperationException($"Unknwon rid format {rid}.");
+
+            string osNameAndVersion = ridParts[0];
+
+            // For now force the rid to always be win7 for windows.
+            if (osNameAndVersion.StartsWith("win"))
+            {
+                osNameAndVersion = "win7";
```

Comment:
Yes I've been debating about whether or not to remove this but I was mostly following @mellinoe's lead here. I'm happy for us to remove this and do the necessary testing and if it works not have this hack.